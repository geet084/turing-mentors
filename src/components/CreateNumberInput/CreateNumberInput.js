import React from 'react';

export const CreateNumberInput = (props) => {
  const { handleChange, max, text, value } = props;
  const name = text.toLowerCase().split(' ')[0]

  return (
    <div className="input-box">
      <input
        className={value ? 'input-effect has-content' : 'input-effect'}
        maxLength={max}
        name={name}
        onChange={handleChange}
        type="text"
        value={value}
      />
      <label>{text}</label>
      <span className="focus-border"></span>
    </div>
  )
}

export default CreateNumberInput;