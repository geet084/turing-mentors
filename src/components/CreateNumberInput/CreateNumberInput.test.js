import React from 'react';
import { shallow } from 'enzyme';
import { CreateNumberInput } from './CreateNumberInput';

describe('CreateNumberInput', () => {
  let wrapper;
  let mockProps = {
    text: '',
    value: '',
    handleChange: jest.fn()
  };

  beforeEach(() => {
    wrapper = shallow(<CreateNumberInput {...mockProps} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should add a class if the input is not empty', () => {
    mockProps = {
      text: 'someText',
      value: 'not empty',
      max: '99',
      handleChange: jest.fn()
    };

    expect(wrapper).toMatchSnapshot();
  });

  it('should remove a class if the input is empty', () => {
    mockProps = {
      text: 'someText',
      value: '',
      max: '99',
      handleChange: jest.fn()
    };

    expect(wrapper).toMatchSnapshot();
  });
});