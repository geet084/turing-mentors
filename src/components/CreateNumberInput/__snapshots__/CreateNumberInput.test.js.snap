// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateNumberInput should add a class if the input is not empty 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateNumberInput
    handleChange={[MockFunction]}
    text=""
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="input-effect"
          name=""
          onChange={[MockFunction]}
          type="text"
          value=""
        />,
        <label>
          
        </label>,
        <span
          className="focus-border"
        />,
      ],
      "className": "input-box",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input-effect",
          "maxLength": undefined,
          "name": "",
          "onChange": [MockFunction],
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "focus-border",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="input-effect"
            name=""
            onChange={[MockFunction]}
            type="text"
            value=""
          />,
          <label>
            
          </label>,
          <span
            className="focus-border"
          />,
        ],
        "className": "input-box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input-effect",
            "maxLength": undefined,
            "name": "",
            "onChange": [MockFunction],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "focus-border",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`CreateNumberInput should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateNumberInput
    handleChange={[MockFunction]}
    text=""
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="input-effect"
          name=""
          onChange={[MockFunction]}
          type="text"
          value=""
        />,
        <label>
          
        </label>,
        <span
          className="focus-border"
        />,
      ],
      "className": "input-box",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input-effect",
          "maxLength": undefined,
          "name": "",
          "onChange": [MockFunction],
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "focus-border",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="input-effect"
            name=""
            onChange={[MockFunction]}
            type="text"
            value=""
          />,
          <label>
            
          </label>,
          <span
            className="focus-border"
          />,
        ],
        "className": "input-box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input-effect",
            "maxLength": undefined,
            "name": "",
            "onChange": [MockFunction],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "focus-border",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`CreateNumberInput should remove a class if the input is empty 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateNumberInput
    handleChange={[MockFunction]}
    max="99"
    text="someText"
    value="not empty"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="input-effect has-content"
          maxLength="99"
          name="sometext"
          onChange={[MockFunction]}
          type="text"
          value="not empty"
        />,
        <label>
          someText
        </label>,
        <span
          className="focus-border"
        />,
      ],
      "className": "input-box",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input-effect has-content",
          "maxLength": "99",
          "name": "sometext",
          "onChange": [MockFunction],
          "type": "text",
          "value": "not empty",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "someText",
        },
        "ref": null,
        "rendered": "someText",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "focus-border",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="input-effect has-content"
            maxLength="99"
            name="sometext"
            onChange={[MockFunction]}
            type="text"
            value="not empty"
          />,
          <label>
            someText
          </label>,
          <span
            className="focus-border"
          />,
        ],
        "className": "input-box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input-effect has-content",
            "maxLength": "99",
            "name": "sometext",
            "onChange": [MockFunction],
            "type": "text",
            "value": "not empty",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "someText",
          },
          "ref": null,
          "rendered": "someText",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "focus-border",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
